txpipe/diagnostics.py

```diff
@@ -678,6 +672,146 @@ def plot_mag_histograms(self):
             fig.close()
 
 
+class TXLensDiagnosticPlots(PipelineStage):
+    """
+    """
+    name='TXLensDiagnosticPlots'
+    # This tells ceci to launch under dask:
+    dask_parallel = True
+
+    inputs = [
+        ('photometry_catalog', HDFFile),
+        ('lens_tomography_catalog', TomographyCatalog),
+    ]
+
+    outputs = [
+        ('lens_snr_hist', PNGFile),
+        ('lens_mag_hist', PNGFile),
+
+    ]
+
+    config_options = {
+        'chunk_rows': 50000,
+        'delta_gamma': 0.02,
+        'mag_min': 18,
+        'mag_max': 28,
+        'snr_min': 5,
+        'snr_max': 200,
+        'bands': 'ugrizy',
+    }
+
+    def run(self):
+        # PSF tests
+        import matplotlib
+        matplotlib.use('agg')
+
+        files, data, nbin = self.load_data()
+        self.plot_snr_histograms(data, nbin)
+        self.plot_mag_histograms(data, nbin)
+
+        # These need to stay open until the end
+        for f in files:
+            f.close()
+
+
+    def load_data(self):
+        import dask.array as da
+
+
+        bands = self.config['bands']
+        # These need to stay open until dask has finished with them.:
+        f = self.open_input("photometry_catalog")
+        g = self.open_input("lens_tomography_catalog")
+
+        # read nbin from metadata
+        nbin = g['tomography'].attrs['nbin_lens']
+
+        # This seems to work okay; it's also possible to set it to a specific count or
+        # to a data size
+        block = 'auto'
+
+        # Load data columns, lazily for dask
+        data = {}
+        for b in bands:
+            data[f'mag_{b}'] = da.from_array(f[f'photometry/mag_{b}'], block)
+            data[f'snr_{b}'] = da.from_array(f[f'photometry/snr_{b}'], block)
+        data['bin'] = da.from_array(g['tomography/lens_bin'], block)
+
+        # Avoid recomputing selections in each histogram by doing it externally here
+        data['sel'] = da.where(data['bin'] >= 0)
+        for i in range(nbin):
+            data[f'sel{i}'] = da.where(data['bin'] >= 0)
+
+        # Return the open files so they stay open until dask has finished with them.
+        # and also the dict of lazy columns and the bin info
+        return [f, g], data, nbin
+
+
+    def plot_snr_histograms(self, data, nbin):
+        smin = self.config['snr_min']
+        smax = self.config['snr_max']
+        bins = np.geomspace(smin, smax, 50)
+        xlog = True
+        self.plot_histograms(data, nbin, 'snr', xlog, bins)
+
+    def plot_mag_histograms(self, data, nbin):
+        import dask.array as da
```

Comment:
why is it imported here, but not in the snr histograms?

Reply:
Thanks - removed